src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultGeneratedDocumentPublisher.cs

```diff
@@ -89,11 +129,19 @@ private void ProjectSnapshotManager_Changed(object sender, ProjectChangeEventArg
             {
                 case ProjectChangeKind.DocumentChanged:
                 case ProjectChangeKind.DocumentRemoved:
-                    if (_publishedSourceText.ContainsKey(args.DocumentFilePath) &&
+                    if (_publishedCSharpSourceText.ContainsKey(args.DocumentFilePath) &&
+                        !_projectSnapshotManager.IsDocumentOpen(args.DocumentFilePath))
```

Comment:
Wasn't there another issue here about not checking open for `DocumentRemoved`?

Reply:
Yeah same issue here as in `DocumentVersionCache` https://github.com/dotnet/aspnetcore-tooling/pull/1607/files#diff-d7d330fb43238c40c3ca2a05d92b5da2R118. I'll fix it.